### PR TITLE
Handlebars3.0.3 upgrade - picking up where chrisrink left off

### PR DESCRIPTION
**Pull request by ryechus**
I started where @chrisrink left off and was able to fix the problems he was having pretty simply.

### CODE REVIEW

File: hbs.js [hunk 1]
```diff
@@ -192,7 +192,7 @@ define([
       }
     },
 
-    version: '2.0.0',
+        version: '3.0.3',
```
Comment (SlexAxton) [inline hbs.js]: Weird spacing?

**Comment by SlexAxton** [comment]
Looks pretty good. Nice tests too!

**Comment by ryechus** [comment]
I removed the extra space. Let me know if you find anything else.

**Comment by SlexAxton** [comment]
Nope. Looks solid.

**Comment by chrisrink** [comment]
Thanks @ryechus  for finishing this off. A few small notes. I actually got this working and have been testing it in my production environment for the last month with the follow:

{code}
 function isHelper(statement){
        return !!((statement.path && statement.path.original) && (statement.type === 'SubExpression' || (statement.params && statement.params.length) || statement.hash));
      }
{code}

So @ryechus solution is pretty much the same thing with the exception of the additional statement.path.original check. I don't remember the use case exactly of when that was used. Nested helper in a partial maybe? Anyways, if that pops up I believe that would be the solution.

In the long term, I made a patch request for Handlebars to better figure this out in their public function
{code}
Handlebar.AST.helpers.helperExpression
{code}

This looks like it will be fixed in Handlebars 4.0 but it might be a good idea to use the one tested by the Handlebars guys. 

{code}
 helperExpression: function(node) {
          return (node.type === 'SubExpression')
                     || ((node.type === 'MustacheStatement' || node.type === 'BlockStatement')
                     && !!((node.params && node.params.length) || node.hash));
{code}
https://github.com/wycats/handlebars.js/pull/1055

Thanks again

**Pull request merged**